zebra/zebra_etcd.c

```diff
@@ -0,0 +1,639 @@
+#include <zebra.h>
+#include <arpa/inet.h>
+#include "etcd_client_txn_wrapper.h"
+#include "etcd_client_wrapper.h"
+#include "log.h"
+#include "libfrr.h"
+#include "stream.h"
+#include "thread.h"
+#include "network.h"
+#include "command.h"
+#include "version.h"
+
+#include "zebra/rib.h"
+#include "zebra/zserv.h"
+#include "zebra/zebra_ns.h"
+#include "zebra/zebra_vrf.h"
+#include "zebra/zebra_errors.h"
+#include "zebra/zebra_fpm_private.h"
+#include "fpm/fpm.h"
+#include "lib/prefix.h"
+
+/*
+ *  * Interval at which we attempt to connect to the ETCD.
+ *   */
+#define ZETCD_CONNECT_RETRY_IVL   5
+#define ZETCD_CONNECT_RETRY_ATTEMPT   5
+
+/*
+ *  * Sizes of outgoing and incoming stream buffers for writing/reading
+ *   * ETCD messages.
+ *    */
+#define ZETCD_OBUF_SIZE (2 * ETCD_MAX_MSG_LEN)
+#define ZETCD_IBUF_SIZE (ETCD_MAX_MSG_LEN)
```

Comment:
At the time of implementation, It was assumed that message will also flow from `etcd server` to `zebra`, to store those messages need a buffer with size `ZETCD_IBUF_SIZE`. But it turns out that it is not the case. Its only one way from `zebra` to `etcd server`. will remove `ZETCD_IBUF_SIZE` macro.
Will use `ZETCD_OBUF_SIZE` macro instead of hard-coded value for `obuf` (i.e. `2*4096`) while declaration.